### ISSUE TITLE
How to use float labels in the training!

### ISSUE THREAD

**Issue by yechaochen** [apache/incubator-mxnet#2915]
I have generate my lst file like:

```
98572   -1.7156 1.257   1200292226/20100406_065150.jpg
98573   -1.7445 1.2286  1200292226/20100406_070056.jpg
98574   -1.7823 1.1922  1200292226/20100406_071240.jpg
98575   -1.8101 1.1661  1200292226/20100406_072106.jpg
98576   -1.8762 1.1067  1200292226/20100406_074035.jpg
98577   -1.9201 1.0694  1200292226/20100406_075301.jpg
98578   -1.9601 1.0371  1200292226/20100406_080359.jpg
```

But when I try to training an error happened:
`AttributeError: 'bool' object has no attribute 'sum'`

I make a experiment that I change my label in lst file like:

```
98572   2     5    1200292226/20100406_065150.jpg
98573   -1    4   1200292226/20100406_070056.jpg
98574   -4    1   1200292226/20100406_071240.jpg

```

And run the training again.The error is gone and training is done.

So the error seems caused by the float labels.Can someone tell me how to use the float labels in mxnet?

**Comment by antinucleon** [comment]
What is the second & third column's meaning? Label is assumed to be non-negative

**Comment by yechaochen** [comment]
@antinucleon But I have experiment in a demo dataset. 
```
1  -3      5       Berkeley/20080506_102001.jpg
2  -5      3       Berkeley/20080506_102501.jpg
3  -6      3       Berkeley/20080506_103001.jpg
4  -1      2       Berkeley/20080506_103501.jpg
5  -2      2       Berkeley/20080506_104001.jpg
```
And the training process only exist when the batch_size set 1.In case of batch_size > 1. The error 'AttributeError: 'bool' object has no attribute 'sum'' occure to me again.

**Comment by antinucleon** [comment]
I still can't get what you intend to do, even don't know it is for classification or regression.

**Comment by yechaochen** [comment]
@antinucleon My labels is float type. And it's both negative and positive in the form of -1.257 and 4.325.
The purpose of my net is generating weo ouputs in the former type at the same time.
But I don't know how to set the lst file and the net in the form mxnet allowed.

**Comment by yechaochen** [comment]
@antinucleon Maybe I don't show my task specific.
I have a dataset contains 10W images and every image has two labels like -1.257 and 4.325.
So I follw the im2rec.py build the lst file like:
'''
98572   -1.7156 1.257   1200292226/20100406_065150.jpg
98573   -1.7445 1.2286  1200292226/20100406_070056.jpg
98574   -1.7823 1.1922  1200292226/20100406_071240.jpg
98575   -1.8101 1.1661  1200292226/20100406_072106.jpg
98576   -1.8762 1.1067  1200292226/20100406_074035.jpg
98577   -1.9201 1.0694  1200292226/20100406_075301.jpg
98578   -1.9601 1.0371  1200292226/20100406_080359.jpg
'''
The I use the tool im2rec transform the lst file to rec file with label_width=2

The purpose of my task is making a prediction of a image and generate the two number like the type in lst file of the image at the same time.

Then I copy 'find_mxnet.py''train_model.py''train_alexnet.py''symbol_googlenet.py' to my folder 
And built my net in 'symbol.mynet.py'.(I have renamed this file) The last layer of my net is:
'''
s2_fc2 = mx.symbol.FullyConnected(data=s2_relu5, num_hidden=2)
linearRegression = mx.symbol.LinearRegressionOutput(data=s2_fc2, name="softmax")
return linearRegression
'''
**Q1----I have searched in google about multi-label task in mxnet and I have notice that the last layer of my net   may be wrong. But I don't know how to write the output layer in the correct type.**

I built my dataiter in 'train_mynet.py' with ImageRecordIter
'''
def get_iterator(args, kv):
    data_shape = (3, args.data_shape, args.data_shape)
    train = mx.io.ImageRecordIter(
        path_imgrec = os.path.join(args.data_dir, args.train_dataset),
        mean_img = "/home/deeper/10W-project/ye_train.bin",
        data_shape  = data_shape,
        path_imglist= "/home/deeper/10W-project/ye_train.lst",
        label_width = 2,
        batch_size  = args.batch_size,
        num_parts   = kv.num_workers,
        part_index  = kv.rank)
    val = mx.io.ImageRecordIter(
        path_imgrec = os.path.join(args.data_dir, args.val_dataset),
        mean_img = "/home/deeper/10W-project/ye_train.bin",
        rand_crop   = False,
        rand_mirror = False,
        data_shape  = data_shape,
        path_imglist= "/home/deeper/10W-project/ye_val.lst",
        label_width = 2,
        batch_size  = args.batch_size,
        num_parts   = kv.num_workers,
        part_index  = kv.rank)

    return (train, val)
'''

**Q2-----All  preparetions of my task is show above. But the training process is going on only when I set the batch_size=1. When I change the batch_size I encounter the error'AttributeError: 'bool' object has no attribute 'sum'' **
